### PR TITLE
Modernize systemd patches

### PR DESCRIPTION
**Pull request by EduardGomezEscandell**
My attempt at modernizing the system patches.

WIP as systemd enablement should not be release-agnostic.

WSL-515

### CODE REVIEW

File: DistroLauncher/systemd_config.cpp [hunk 1]
```diff
@@ -1,73 +1,17 @@
...
     /**
      * Disables systemd-binfmt.service to avoid breaking interop.
      * See related bug report: https://github.com/ubuntu/WSL/issues/334
      */
-    bool MaskSystemdBinfmtService() noexcept
+    bool MaskBinfmtService() noexcept
     {
         DWORD exitCode;
         auto hr =
           Sudo::WslLaunchInteractive(L"ln -s /dev/null /etc/systemd/system/systemd-binfmt.service", FALSE, &exitCode);
```
[the remark this answers is not in the document]
Comment (EduardGomezEscandell) [reply DistroLauncher/systemd_config.cpp:14]: Come to think of it, why not simply use the `ConditionVirtualization` trick?

[the remark this answers is not in the document]
Comment (EduardGomezEscandell) [reply DistroLauncher/systemd_config.cpp]: Just tested it, it works.

We can get rid of all the old code :D.

[the remark this answers is not in the document]
Comment (CarlosNihelton) [reply DistroLauncher/systemd_config.cpp]: Lovely!

**Review by CarlosNihelton — APPROVED** [review]
This looks great!

**Comment by EduardGomezEscandell** [comment]
Rebased to solve conflicts with after #382

**Comment by CarlosNihelton** [comment]
@EduardGomezEscandell, I'll leave the last two commits for you to consider. It looks like we can error early with `Access denied` by running any other WSL command before `wsl --unregister` and manage to proceed with the CI.

image

**Comment by EduardGomezEscandell** [comment]
That is some weird behaviour. I suppose it is fine if that leads to the CI functioning afterwards.

**Comment by EduardGomezEscandell** [comment]
Forced-push to rebase main.

**Comment by CarlosNihelton** [comment]
As before, I took the freedom to rebase it to get the CI all green and finally merge it.

**Pull request merged**